compiler-rt/lib/radsan/radsan.h

```diff
@@ -8,17 +8,38 @@
 
 #pragma once
 
-#define RADSAN_EXPORT __attribute__((visibility("default")))
+#include "sanitizer_common/sanitizer_atomic.h"
+#include "sanitizer_common/sanitizer_internal_defs.h"
+#include "radsan_flags.h"
+
+namespace radsan {
+
+#define ENSURE_RADSAN_INITED() do { \
+  using namespace radsan; \
+  CHECK(!radsan_init_is_running); \
+  if (!radsan_inited) { \
+    radsan_init(); \
+  } \
+} while (0)
+
+// TODO: are these data races?? Things may be initialized in multiple threads
```

Comment:
Eyes here, I think there is a big possibility these bools could be data races. msan doesn't have them as atomics, but I am suspicious 🤔

Reply:
I opted to change these to static and atomic for safety reasons. I don't think msan is doing it right..

Reply:
Interesting. I'm not sure. What's your thinking?